### PR TITLE
manifest: Update upmerged respositories

### PR DESCRIPTION
**Pull request by de-nordic**
Upmerging:
  - sdk-mcuboot to 0c0470e294dcfb52aab92299356a5f3caa0aa52b
  - sdk-zephyr to 23cf38934c0f68861e403b22bc3dd0ce6efbfa39
  
test-sdk-nrf: test_upmerge_nov_23

### CODE REVIEW

**Review by sigvartmh — APPROVED** [review]

**Review by eivindj-nordic — APPROVED** [review]

**Review by tony-le-24 — APPROVED** [review]

**Review by jayteemo — APPROVED** [review]

**Review by plskeggs — APPROVED** [review]

**Review by glarsennordic — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

**Review by jtguggedal — APPROVED** [review]

**Review by juhaylinen — APPROVED** [review]

**Comment by aescolar** [comment]
Fixes for the asset_tracker_v2 samples/tests in https://github.com/aescolar/sdk-nrf/commits/asset_tracker_fixes 
Fill free to pick them into this branch

**Comment by de-nordic** [comment]
> https://github.com/aescolar/sdk-nrf/commits/asset_tracker_fixes

Picked. Thanks

Next time please do PR to my PR.

**Review by canisLupus1313 — APPROVED** [review]

**Review by MarkusLassila — APPROVED** [review]

**Comment by aescolar** [comment]
@MarekPieta In https://github.com/nrfconnect/sdk-nrf/pull/12619/commits/4ff3f2b1fa9ad0668b8de607dbad39486dae597d did you intend to re-define NRF52_ANOMALY_109_WORKAROUND_EGU_INSTANCE without a type? (note that fails when the nrf52/Kconfig.soc is not included as that option lacks a type then.

**Review by MarekPieta — APPROVED** [review]
Seems ok for our applications (Find My, nRF Desktop, Fast Pair)

**Comment by RobertGalatNordic** [comment]
Is there some changelog ? It seems like a huge change, 500 files have been modified. It would be nice to see some bullet points with the major changes that we should know about

File: subsys/bluetooth/gatt_dm.c [hunk 1]
```diff
@@ -13,7 +13,7 @@
 LOG_MODULE_REGISTER(bt_gatt_dm, CONFIG_BT_GATT_DM_LOG_LEVEL);
 
 /* Available sizes: 128, 512, 2048... */
-#define CHUNK_DATA_SIZE (128 - sizeof(struct k_mem_block_id) \
+#define CHUNK_DATA_SIZE (128 - sizeof(struct k_heap *) \
```
Comment (pdunaj) [inline subsys/bluetooth/gatt_dm.c:16]: Should be ok but please check @grochu.

**Comment by Damian-Nordic** [comment]
Would it make sense to keep the `hci_rpmsg` child image name despite of the Zephyr sample name change? I'm not sure if the rename brings much value given the amount of work it will impose on all our customers (and the number of conflicts it will generate).

**Comment by nordicjm** [comment]
> Would it make sense to keep the `hci_rpmsg` child image name despite of the Zephyr sample name change? I'm not sure if the rename brings much value given the amount of work it will impose on all our customers (and the number of conflicts it will generate).

No.

File: drivers/sensor/paw3212/paw3212.c [hunk 2]
```diff
@@ -548,7 +548,7 @@ static void trigger_handler(struct k_work *work)
 	sensor_trigger_handler_t handler;
 	int err = 0;
 	struct paw3212_data *data = CONTAINER_OF(work, struct paw3212_data,
-						     trigger_handler_work);
+						 trigger_handler_work);
```
[the remark this answers is not in the document]
Comment (nordicjm) [reply drivers/sensor/paw3212/paw3212.c:551]: @MarekPieta

**Comment by Damian-Nordic** [comment]
> > Would it make sense to keep the `hci_rpmsg` child image name despite of the Zephyr sample name change? I'm not sure if the rename brings much value given the amount of work it will impose on all our customers (and the number of conflicts it will generate).
> 
> No.

Thank you for your comprehensive and respectful response. May I ask why though?

File: samples/peripheral/802154_phy_test/prj.conf [hunk 3]
```diff
@@ -16,8 +16,8 @@ CONFIG_NRFX_POWER=y
 
 # Enable RNG
 CONFIG_NRFX_RNG=y
-# Use XOROSHIRO PRNG
-CONFIG_XOROSHIRO_RANDOM_GENERATOR=y
```
Comment (jciupis) [inline samples/peripheral/802154_phy_test/prj.conf:20]: What's the reason for this change?

**Review by jciupis — APPROVED** [review]

File: samples/peripheral/802154_phy_test/prj.conf (hunk 3 above)
Comment (nordicjm) [reply samples/peripheral/802154_phy_test/prj.conf:20]: https://docs.zephyrproject.org/latest/releases/release-notes-3.5.html

> CONFIG_XOROSHIRO_RANDOM_GENERATOR, deprecated a long time ago, is finally removed.

File: sysbuild/CMakeLists.txt [hunk 4]
```diff
@@ -186,8 +186,10 @@ function(${SYSBUILD_CURRENT_MODULE_NAME}_pre_cmake)
       )
     endif()
 
-    # Use NCS signing script with support for PM
-    set(${DEFAULT_IMAGE}_SIGNING_SCRIPT "${ZEPHYR_NRF_MODULE_DIR}/cmake/sysbuild/image_signing.cmake" CACHE INTERNAL "MCUboot signing script" FORCE)
+    if(SB_CONFIG_PARTITION_MANAGER)
+      # Use NCS signing script with support for PM
+      set(${DEFAULT_IMAGE}_SIGNING_SCRIPT "${ZEPHYR_NRF_MODULE_DIR}/cmake/sysbuild/image_signing.cmake" CACHE INTERNAL "MCUboot signing script" FORCE)
```
Comment (tejlmand) [inline sysbuild/CMakeLists.txt:191]: minor nit. the use of `INTERNAL` implies `FORCE` so `FORCE` can be stripped.

(I do notice that the original code has same use of `FORCE`)

**Review by tejlmand — APPROVED** [review]

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst [hunk 5]
```diff
@@ -30,6 +30,18 @@ The following changes are mandatory to make your application work in the same wa
       * The ``CONFIG_CHIP_SED_ACTIVE_INTERVAL`` Kconfig option was renamed to :kconfig:option:`CONFIG_CHIP_ICD_FAST_POLLING_INTERVAL`.
       * The ``CONFIG_CHIP_SED_ACTIVE_THRESHOLD`` Kconfig option was renamed to :kconfig:option:`CONFIG_CHIP_ICD_ACTIVE_MODE_THRESHOLD`.
 
+* For samples using Wi-Fi features:
+
+  * A few Kconfig options related to scan operations have been removed in the current release. If your application uses them,
+    they will need to be updated to remove the dependency on these:
```
Comment (greg-fer) [inline doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst:36]:
```suggestion
  * A few Kconfig options related to scan operations have been removed in the current release. If your application uses scan operations, they need to be updated to remove the dependency on the following options:
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 6]
```diff
@@ -211,6 +211,10 @@ nRF Desktop
   * Changed the term *flash memory* to *non-volatile memory* for generalization purposes.
   * The :ref:`nrf_desktop_usb_state` to use the :c:func:`usb_hid_set_proto_code` function to set the HID Boot Interface protocol code.
     The ``CONFIG_USB_HID_PROTOCOL_CODE`` Kconfig option is deprecated and a dedicated API needs to be used instead.
+  * The :ref:`nrf_desktop_usb_state` to use :c:func:`usb_hid_set_proto_code` function to set the HID Boot Interface protocol code.
+    The ``CONFIG_USB_HID_BOOT_PROTOCOL`` Kconfig option was removed and dedicated API needs to be used instead.
+  * Disabled MCUboot's logs over RTT (:kconfig:option:`CONFIG_LOG_BACKEND_RTT` and :kconfig:option:`CONFIG_USE_SEGGER_RTT`) on ``nrf52840dk_nrf52840`` in :file:`prj_mcuboot_qspi.conf` configuration to reduce MCUboot memory footprint and avoid flash overflows.
+    Explicitly enabled the UART log backend (:kconfig:option:`CONFIG_LOG_BACKEND_UART`) together with its dependencies in the configuration file to ensure log visibility.
```
Comment (greg-fer) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:217]: @MarekPieta , can you check these lines for accuracy?

**Review by greg-fer — CHANGES_REQUESTED** [review]

**Comment by greg-fer** [comment]
(Requested changes not applied yet.)

File: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst (hunk 5 above)
Comment (nordicjm) [reply doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst]: Updates

File: drivers/sensor/paw3212/paw3212.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (MarekPieta) [reply drivers/sensor/paw3212/paw3212.c:551]: There is no need for such change. `trigger_handler_work` is a `k_work` (not `k_work_delayable`) so that type of `work` pointer already matches it.
For delayable works, a regular work item (that is member of delayable work structure) is passed as work handler argument, so we need to add `.work` to access it.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 6 above)
Comment (MarekPieta) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:224]: The first bullet point here seems duplicated. We should remove it. Apart from it looks good :)
@doer-nordic can we remove it, please?

Comment (de-nordic) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:224]: Will fix

**Review by alstrzebonski — APPROVED** [review]

**Review by trantanen — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]
Checked .rst, .h, and .txt files. Reported a nit to @divipillai in asset_tracker_v2_description that can be handled in a separate PR.

**Comment by nordicjm** [comment]
Typo in last commit will be fixed when other repos are merged

File: subsys/bluetooth/gatt_dm.c (hunk 1 above)
Comment (pdunaj) [reply subsys/bluetooth/gatt_dm.c:16]: ping? or maybe @KAGA164

**Review by pdunaj — APPROVED** [review]

**Comment by de-nordic** [comment]
> could we cherry-pick these into sdk-zephyr from upstream? Would be super helpful for thingy91x board files:
> 
>     * drivers: regulator: Added startup and off/on delay to common driver `04e18f093fde6ed2671f52c3f373edf546259783`
> 
>     * soc: arm: nordic_nrf: nrf91: add nRF9151 LACA `84f4ffce7c217b3dd4c9b4dc8044f6220775a020`
> 
>     * boards: arm: add nrf9151dk_nrf9151 `c055561a6b12dd530fa54452be6e80d138dfbd50`

If it does not fix anything within upmerge we can just pick them when upmerge concludes.

**Review by oyvindronningstad — APPROVED** [review]

**Review by shanthanordic — APPROVED** [review]

**Review by magnev — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (balaji-nordic) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:1]: Do we plan to update the migration guide too after this PR is in? 

CC: @umapraseeda

File: applications/asset_tracker_v2/doc/unit_test.rst
Comment (balaji-nordic) [inline applications/asset_tracker_v2/doc/unit_test.rst:1]: @aescolar Any particular reason for migrating the asset tracker tests to native_sim ? Its a good thing to do of course. but was that necessary in this PR? There are some other tests (like those under `tests/subsys/net`) that are still running on native_posix board instead of native_sim. Should those be updated too?

File: applications/nrf5340_audio/src/audio/Kconfig.defaults [hunk 7]
```diff
@@ -16,10 +16,6 @@ config NRFX_DPPI
 config CMSIS_DSP
 	default y
 
-# Required for calculating test tones
-config CMSIS_DSP_FASTMATH
```
Comment (koffes) [inline applications/nrf5340_audio/src/audio/Kconfig.defaults:20]: Whay was this removed?

Comment (nordicjm) [reply applications/nrf5340_audio/src/audio/Kconfig.defaults:20]: >     treewide: Remove dependency on no longer existing Kconfigs
>     
>     The Zephyr commit 104ea2ccca1f62048fcc9535e182809ffdf6fd8b
>     removed:
>      CONFIG_CMSIS_DSP_FASTMATH
>      CONFIG_CMSIS_DSP_TABLES

File: tests/nrf5340_audio/sw_codec_lc3/prj.conf [hunk 8]
```diff
@@ -3,8 +3,6 @@ CONFIG_IRQ_OFFLOAD=y
 CONFIG_FPU=y
 CONFIG_NEWLIB_LIBC=y
 CONFIG_CMSIS_DSP=y
-CONFIG_CMSIS_DSP_FASTMATH=y
-CONFIG_CMSIS_DSP_TABLES=y
```
Comment (koffes) [inline tests/nrf5340_audio/sw_codec_lc3/prj.conf:7]: See earlier comment

**Review by Frodevan — CHANGES_REQUESTED** [review]
Almost good for audio test, but there is a minor discrepancy in the kconfig for one setting in one particular build-condfiguration of the audio application.

**Review by Frodevan — APPROVED** [review]
Ok, CONFIG_NCS_INCLUDE_RPMSG_CHILD_IMAGE may still be called as such and not CONFIG_NCS_INCLUDE_IPC_CHILD_IMAGE, but we have decided that this is not a blocker for the Audio project as of now.
